### PR TITLE
0.1.0/types

### PR DESCRIPTION
**Pull request by MainShayne233**
This adds support and tests for the supported types outlines in `0.1.0.md`

### CODE REVIEW

File: lib/breakfast/type.ex [hunk 1]
```diff
@@ -1,18 +1,51 @@
...
+    float: {Kernel, :is_float},
+    number: {Kernel, :is_number},
+    atom: {Kernel, :is_atom},
+    tuple: {Kernel, :is_tuple},
+    list: {Kernel, :is_list},
+    module: {Kernel, :is_atom}
   }
 
-  @understood_primative_types Map.keys(@understood_primative_type_predicate_mappings)
+  @understood_primative_types Map.keys(@understood_primative_type_predicate_mappings) ++
```
Comment (evuez) [inline lib/breakfast/type.ex]:
```suggestion
  @understood_primitive_types Map.keys(@understood_primitive_type_predicate_mappings) ++
```

File: lib/breakfast/type.ex [hunk 2]
```diff
@@ -1,18 +1,51 @@
 defmodule Breakfast.Type do
   @moduledoc false
   alias TypeReader.TerminalType
+  alias Breakfast.Field
 
   @understood_primative_type_predicate_mappings %{
```
Comment (evuez) [inline lib/breakfast/type.ex]:
```suggestion
  @understood_primitive_type_predicate_mappings %{
```

File: lib/breakfast/type.ex [hunk 3]
```diff
@@ -102,11 +208,73 @@ defmodule Breakfast.Type do
...
+    if is_integer(term) and term >= min and term <= max do
+      []
+    else
+      ["expected an integer in #{min}..#{max}, got: #{term}"]
+    end
+  end
 
-  for {type, predicate} <- @understood_primative_type_predicate_mappings do
+  for {type, {predicate_module, predicate_function}} <-
+        @understood_primative_type_predicate_mappings do
```
Comment (evuez) [inline lib/breakfast/type.ex]:
```suggestion
        @understood_primitive_type_predicate_mappings do
```

File: test/breakfast_test.exs [hunk 4]
```diff
@@ -520,4 +519,209 @@ defmodule BreakfastTest do
...
+                 literal_struct:
+                   "expected a %Breakfast.TestDefinitions.Struct{}, got: %Breakfast.TestDefinitions.OtherStruct{email: \"\"}",
+                 literal_typed_struct:
+                   "expected a field with key :name and value of type :binary, got: invalid value: [\"expected a binary, got: 42\"]",
+                 literal_empty_tuple: "expected {}, got: {:apples}",
+                 literal_typed_tuple:
+                   "expected {:atom, :binary, :integer}, got: {\"apples\", \"oranges\", 123.56}",
+                 union_type: "expected one of :integer | :never | :infinity, got: :always",
+                 remote: "expected one of :red | :green | :blue, got: :purple"
+               ]
```
Comment (evuez) [inline test/breakfast_test.exs:724]: :heart: :tada:

**Review by evuez — APPROVED** [review]

**Comment by evuez** [comment]
@MainShayne233 LGTM, just pushed a commit to fix a typo in `primitive`

**Comment by MainShayne233** [comment]
@evuez ❤️

**Pull request merged**